### PR TITLE
[AArch64] Implement intrinsics for F1CVTL/F2CVTL and BF1CVTL/BF2CVTL

### PR DESCRIPTION
**Pull request by SpencerAbson**
This patch implements the following intrinsics:

8-bit floating-point convert to deinterleaved half-precision or BFloat16.
``` c
  // Variant is also available for: _bf16[_mf8]_x2
  svfloat16x2_t svcvtl1_f16[_mf8]_x2_fpm(svmfloat8_t zn, fpm_t fpm) __arm_streaming;
  svfloat16x2_t svcvtl2_f16[_mf8]_x2_fpm(svmfloat8_t zn, fpm_t fpm) __arm_streaming;
```

Defined in https://github.com/ARM-software/acle/pull/323

Co-authored-by: Caroline Concatto caroline.concatto@arm.com
Co-authored-by: Marian Lukac marian.lukac@arm.com

### CODE REVIEW

File: llvm/test/CodeGen/AArch64/sme2-fp8-intrinsics-cvt.ll [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+; CHECK-LABEL: bf2cvtl:
+; CHECK:       // %bb.0:
+; CHECK-NEXT:    bf2cvtl { z0.h, z1.h }, z0.b
+; CHECK-NEXT:    ret
+  %res = call { <vscale x 8 x bfloat>, <vscale x 8 x bfloat> } @llvm.aarch64.sme.fp8.bf2cvtl.x2(<vscale x 16 x i8> %zm)
+  ret { <vscale x 8 x bfloat>, <vscale x 8 x bfloat> } %res
+}
+
+
+declare { <vscale x 8 x half>, <vscale x 8 x half> } @llvm.aarch64.sme.fp8.f1cvtl.x2(<vscale x 16 x i8>)
```
[the remark this answers is not in the document]
Comment (SpencerAbson) [reply llvm/test/CodeGen/AArch64/sme2-fp8-intrinsics-cvt.ll:45]: Good point.

File: clang/lib/CodeGen/CGBuiltin.cpp [hunk 2]
```diff
@@ -11182,6 +11182,10 @@ Value *CodeGenFunction::EmitAArch64SMEBuiltinExpr(unsigned BuiltinID,
            BuiltinID == SME::BI__builtin_sme_svstr_za)
     return EmitSMELdrStr(TypeFlags, Ops, Builtin->LLVMIntrinsic);
 
+  // Emit set FPMR for intrinsics that require it
```
[the remark this answers is not in the document]
Comment (SpencerAbson) [reply clang/lib/CodeGen/CGBuiltin.cpp:11185]: Sure, I'll take a look.

**Review by momchil-velikov — APPROVED** [review]
LGTM, cheers!

**Review by jthackray — APPROVED** [review]
LGTM

**Comment by momchil-velikov** [comment]
We would also need TargetGuard tests, e.g. `/clang/test/Sema/aarch64-sme2-intrinsics`.

**Review by CarolineConcatto — APPROVED** [review]
Thank you Spencer!
LGTM!

**Pull request merged**